### ISSUE TITLE
go/types: not enough arguments error message should be easier to skim

### ISSUE THREAD

**Issue by rsc** [golang/go#26249]
Another difference between gc and go/types where I believe gc ends up more readable:

```
go test -short
# regexp [regexp.test]
./regexp.go:197:20: not enough arguments in call to progMachine
	have (*syntax.Prog, *onePassProg)
	want (*Regexp, *syntax.Prog, *onePassProg)
# regexp
./regexp.go:197:21: cannot use prog (variable of type *regexp/syntax.Prog) as *Regexp value in argument to progMachine
./regexp.go:197:27: cannot use onepass (variable of type *onePassProg) as *regexp/syntax.Prog value in argument to progMachine
./regexp.go:197:34: too few arguments in call to progMachine
vet: typecheck failures
```

**Comment by griesemer** [comment]
Slightly more involved than what we want to do this late in the cycle. Moving to 1.13.

**Comment by griesemer** [comment]
This has been addressed in Go 1.18. Closing.
